contracts/AaveStrategy.sol

```diff
@@ -291,12 +301,28 @@ contract AaveStrategy is CommonContract {
         }
     }
 
-    function swapRewards() internal {
+    function swapRewards(uint _maticToWantRatio) internal {
         uint balance = IERC20(wmatic).balanceOf(address(this));
 
         if (balance > 0) {
+            // _maticToWantRatio is a 9 decimals ratio number calculated by the
+            // caller before call harvest to get the minimum amount of want-tokens.
+            // So the balance is multiplied by the ratio and then divided by 9 decimals
+            // to get the same "precision". Then the result should be divided for the
+            // decimal diff between tokens.
+            // E.g want is USDT with  only 6 decimals:
+            // _maticToWantRatio = 1_522_650_000 (1.52265 USDT/MATIC)
+            // balance = 1e18 (1.0 MATIC)
+            // tokenDiffPrecision = 1e12 (1e18 MATIC decimals / 1e6 USDT decimals)
+            // expected = 1522650 (1e18 * 1_522_650_000 / 1e9 / 1e12) [1.52 in USDT decimals]
+
+            uint tokenDiffPrecision = ERC20(wmatic).decimals().div(
+                ERC20(want).decimals()
+            );
+            uint expected = balance.mul(_maticToWantRatio).div(1e9).div(tokenDiffPrecision);
```

Comment:
You can not have decimal values in `uint`. Therefore, multiply the 9 decimal precision before doing any divisions.
```suggestion
            uint tokenDiffPrecision = ERC20(wmatic).decimals().mul(1e9).div(
                ERC20(want).decimals()
            );
            uint expected = balance.mul(_maticToWantRatio).div(tokenDiffPrecision);
```

Reply:
Checking this again... the tokenDiffPrecision should be the real precision, not the number of decimals, my bad. This should be the correct version, what do you think?:
```solidity
            uint tokenDiffPrecision = (10 ** ERC20(wmatic).decimals()).div(
                10 ** ERC20(want).decimals()
            ).mul(1e9);
            uint expected = balance.mul(_maticToWantRatio).div(tokenDiffPrecision);
```

Reply:
Seems fine, just make sure to use the same method when calculating `_maticToWantRatio`